src/GSF.CarbonAware/src/Handlers/IEmissionsHandler.cs

```diff
@@ -1,15 +1,51 @@
+        /// <returns>Array of EmissionsData objects that contains the location, time and the rating in g/kWh</returns>
+        Task<IEnumerable<EmissionsData>> GetEmissionsDataAsync(string location, DateTimeOffset? start = null, DateTimeOffset? end = null);
+
+        /// <summary>
+        /// Calculate the observed emission data by list of locations for a specified time period.
+        /// </summary>
+        /// <param name="locations">String array of named locations, must be non-empty (ex: ["eastus"])</param>
+        /// <param name="start">[Optional] Start time for the data query. (ex: 2022-03-01T15:30:00Z)</param>
+        /// <param name="end">[Optional] End time for the data query. (ex: 2022-03-01T18:30:00Z)</param>
+        /// <returns>Array of EmissionsData objects that contains the location, time and the rating in g/kWh</returns>
+        Task<IEnumerable<EmissionsData>> GetEmissionsDataAsync(string[] locations, DateTimeOffset? start = null, DateTimeOffset? end = null);
+
+        /// <summary>
+        /// Calculate the best emission data by list of locations for a specified time period.
+        /// </summary>
+        /// <param name="locations">String array of named locations, must be non-empty (ex: ["eastus"])</param>
+        /// <param name="start">[Optional] Start time for the data query. (ex: 2022-03-01T15:30:00Z)</param>
+        /// <param name="end">[Optional] End time for the data query. (ex: 2022-03-01T18:30:00Z)</param>
+        /// <returns>An array of EmissionsData objects that contains the location, time and the rating in g/kWh</returns>
+        Task<IEnumerable<EmissionsData>> GetBestEmissionsDataAsync(string[] locations, DateTimeOffset? start = null, DateTimeOffset? end = null);
+
+        /// <summary>
+        /// Given a location, start, and end time, retrieves the measured carbon intensity data for the given location between the time boundaries and calculates the average carbon intensity during that period. 
         /// </summary>
         /// <remarks> This function is useful for reporting the measured carbon intensity for a specific time period in a specific location. </remarks>
         /// <param name="location">The location name where workflow is run (ex: eastus)</param>
-        /// <param name="start">The time at which the workflow we are measuring carbon intensity for started (ex: 2022-03-01T15:30:00Z)</param>
-        /// <param name="end">The time at which the workflow we are measuring carbon intensity for ended (ex: 2022-03-01T18:30:00Z)</param>
-        /// <returns>The average carbon intensity value.</returns>
-        Task<double> GetAverageCarbonIntensityAsync(string location, DateTimeOffset start, DateTimeOffset end);
+        /// <param name="requestedTimeRange"> A TimeRange object that contains a "startTime" and "endTime" for which to calculate average marginal carbon intensity. </param>
```

Comment:
NIT: use `cref` to link to models

https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/xmldoc/recommended-tags#cref-attribute